rules/0132.go

```diff
@@ -0,0 +1,201 @@
+			},
+		},
+	}
+}
+
+// The List standard method should contain a parent field.
+func checkListRequestMessageParentField() lint.Rule {
+	return &descriptor.CallbackRule{
+		RuleInfo: lint.RuleInfo{
+			Name:         lint.NewRuleName("core", "0132", "request_message", "parent_field"),
+			Description:  "check that a parent field is present",
+			RequestTypes: []lint.RequestType{lint.ProtoRequest},
+			URI:          "https://aip.dev/132#request-message",
+		},
+		Callback: descriptor.Callbacks{
+			MethodCallback: func(m p.MethodDescriptor, s lint.DescriptorSource) (problems []lint.Problem, err error) {
+				// We only care about List- methods for the purpose of this rule;
+				// ignore everything else.
+				if !isListMethod(m) {
+					return
+				}
+
+				// Rule check: Establish that a `parent` field is present.
+				parentField := m.Input().Fields().ByName("parent")
+				if parentField == nil {
+					problems = append(problems, lint.Problem{
+						Message:    fmt.Sprintf("method %q has no `parent` field", m.Name()),
+						Descriptor: m.Input(),
+					})
+					return problems, nil
+				}
+
+				// Rule check: Establish that the parent field is a string.
+				if parentField.Kind() != p.StringKind {
+					problems = append(problems, lint.Problem{
+						Message:    "`parent` field on List RPCs should be a string",
+						Descriptor: parentField,
+					})
+				}
+
+				return problems, nil
+			},
+		},
+	}
+}
+
+// List methods should not have unrecognized fields.
+func checkListRequestMessageUnknownFields() lint.Rule {
+	return &descriptor.CallbackRule{
+		RuleInfo: lint.RuleInfo{
+			Name:         lint.NewRuleName("core", "0132", "request_message", "unknown_fields"),
+			Description:  "check that there are no unknown fields",
+			RequestTypes: []lint.RequestType{lint.ProtoRequest},
+			URI:          "https://aip.dev/132#request-message",
+		},
+		Callback: descriptor.Callbacks{
+			MethodCallback: func(m p.MethodDescriptor, s lint.DescriptorSource) (problems []lint.Problem, err error) {
+				// We only care about List- methods for the purpose of this rule;
+				// ignore everything else.
+				if !isListMethod(m) {
+					return
+				}
+
+				// Rule check: Establish that there are no unexpected fields.
+				allowedFields := map[string]struct{}{
+					"parent":       {}, // AIP-131
+					"page_size":    {}, // AIP-158
+					"page_token":   {}, // AIP-158
+					"filter":       {}, // AIP-132
+					"order_by":     {}, // AIP-132
+					"group_by":     {}, // Nowhere yet, but permitted
+					"show_deleted": {}, // AIP-135
+					"read_mask":    {}, // AIP-157
+					"view":         {}, // AIP-157
+				}
+				fields := m.Input().Fields()
+				for i := 0; i < fields.Len(); i++ {
+					field := fields.Get(i)
+					if _, ok := allowedFields[string(field.Name())]; !ok {
+						problems = append(problems, lint.Problem{
+							Message:    "List RPCs should only contain fields explicitly described in AIPs.",
+							Descriptor: field,
```

Comment:
Do we all agree that a rule should be disabled at the place the warning is shown up?

Reply:
Yes.